### PR TITLE
Np 47972 Refactor test data setup

### PR DESCRIPTION
**Pull request by monaullah**
No change in production code, just an effort to simplify test data setup (reflected in number of lines removed)
- Added new test class `TestChannel` with methods to be represented as a channel registry response body or expected proxy response dto
- Moved duplicate methods to TestUtils
- Modified `ChannelRegistryBodyBuilder` to `ChannelRegistrySearchResponseBodyBuilder`

### CODE REVIEW

File: src/test/java/no/sikt/nva/pubchannels/handler/create/journal/CreateJournalHandlerTest.java [hunk 1]
```diff
@@ -328,28 +332,28 @@ void shouldThrowUnauthorizedIfNotUser() throws IOException {
...
                      dtoObjectMapper.writeValueAsString(request));
         stubFetchResponse(expectedPid);
     }
 
-    private void setupBadRequestStub(String expectedPid, ChannelRegistryCreateJournalRequest request,
-                                     int clientAuthResponseHttpCode,
-                                     int clientResponseHttpCode) throws JsonProcessingException {
-        stubAuth(clientAuthResponseHttpCode);
-        stubResponse(clientResponseHttpCode, "/createjournal/createpid", dtoObjectMapper.writeValueAsString(PROBLEM),
+    private void setupBadRequestStub(String expectedPid, ChannelRegistryCreateJournalRequest request)
```
Comment (LarsV123) [inline src/test/java/no/sikt/nva/pubchannels/handler/create/journal/CreateJournalHandlerTest.java:352]: Linter warning: "Value of parameter 'expectedPid' is always 'null' "

File: src/test/java/no/sikt/nva/pubchannels/handler/create/publisher/CreatePublisherHandlerTest.java [hunk 2]
```diff
@@ -317,32 +322,30 @@ private static Stream<String> invalidIsbnPrefixes() {
...
                      dtoObjectMapper.writeValueAsString(request));
         stubFetchResponse(expectedPid);
     }
 
-    private void setupBadRequestStub(String expectedPid, ChannelRegistryCreatePublisherRequest request,
-                                     int clientAuthResponseHttpCode,
-                                     int clientResponseHttpCode) throws JsonProcessingException {
-        stubAuth(clientAuthResponseHttpCode);
-        stubResponse(clientResponseHttpCode, "/createpublisher/createpid", dtoObjectMapper.writeValueAsString(PROBLEM),
+    private void setupBadRequestStub(String expectedPid, ChannelRegistryCreatePublisherRequest request)
```
Comment (LarsV123) [inline src/test/java/no/sikt/nva/pubchannels/handler/create/publisher/CreatePublisherHandlerTest.java:344]: Linter warning: "Value of parameter 'expectedPid' is always 'null' "

File: src/test/java/no/sikt/nva/pubchannels/handler/create/series/CreateSeriesHandlerTest.java [hunk 3]
```diff
@@ -363,32 +368,30 @@ void shouldThrowUnauthorizedIfNotUser() throws IOException {
...
                      dtoObjectMapper.writeValueAsString(request));
         stubFetchResponse(expectedPid);
     }
 
-    private void setupBadRequestStub(String expectedPid, ChannelRegistryCreateSeriesRequest request,
-                                     int clientAuthResponseHttpCode,
-                                     int clientResponseHttpCode) throws JsonProcessingException {
-        stubAuth(clientAuthResponseHttpCode);
-        stubResponse(clientResponseHttpCode, "/createseries/createpid", dtoObjectMapper.writeValueAsString(PROBLEM),
+    private void setupBadRequestStub(String expectedPid, ChannelRegistryCreateSeriesRequest request)
```
Comment (LarsV123) [inline src/test/java/no/sikt/nva/pubchannels/handler/create/series/CreateSeriesHandlerTest.java:390]: Linter warning: "Value of parameter 'expectedPid' is always 'null' "

File: src/test/java/no/sikt/nva/pubchannels/handler/TestChannel.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        this.name = name;
+        this.onlineIssn = onlineIssn;
+        this.printIssn = printIssn;
+        this.isbnPrefix = isbnPrefix;
+        this.scientificValue = scientificValue;
+        this.discontinued = discontinued;
+        this.sameAs = sameAs;
+    }
+
+    public TestChannel(Integer year, String identifier) {
```
Comment (LarsV123) [inline src/test/java/no/sikt/nva/pubchannels/handler/TestChannel.java:47]: This change replaces the call to "randomIssn" with a random string. Could that cause problems? Do we have (or should we have) tests that expect valid ISSNs etc in generated test data?

**Review by LarsV123 — APPROVED** [review]

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: src/main/java/no/sikt/nva/pubchannels/channelregistry/model/ChannelRegistrySeries.java [hunk 5]
```diff
@@ -16,15 +17,17 @@ public record ChannelRegistrySeries(@JsonProperty(IDENTIFIER_FIELD) String ident
...
+                                                                                                JsonSerializable {
 
     private static final String IDENTIFIER_FIELD = "pid";
     private static final String NAME_FIELD = "originalTitle";
     private static final String ONLINE_ISSN_FIELD = "eissn";
     private static final String PRINT_ISSN_FIELD = "pissn";
     private static final String LEVEL_FIELD = "levelElementDto";
     private static final String HOMEPAGE_FIELD = "kurl";
-    public static final String DISCONTINUED = "ceased";
+    private static final String DISCONTINUED = "ceased";
```
[the remark this answers is not in the document]
Comment (monaullah) [reply src/main/java/no/sikt/nva/pubchannels/channelregistry/model/ChannelRegistrySeries.java:30]: Will do - when I implement new field `isUnderReview`

**Pull request merged**